### ISSUE TITLE
How to remove the managed cluster on hub cluster

### ISSUE THREAD

**Issue by USER0308** [open-cluster-management-io/clusteradm#302]
This is my clusters information:
Hub cluster: 1.19.0 deployed by kubeadm
Managed cluster: 1.19.0 deployed by kubeadm 

The managed cluster has already joined into the hub cluster and I have created some manifectworks in hub cluster. All the k8s resources also created in managed cluster successfully.

Now I am trying to remove the managed cluster from hub. I get two ways from this official article. The first one is running `kubectl delete managedcluster <clusterName>` directly on hub cluster. The second one is stopping klusterlet agent on managed cluster.
And I find the third method from OCM github issue and `clusteradm` command flag which is using `clusteradm unjoin` command on managed cluster.

My first try：
When I try to delete klusterlet pod in namespace `open-cluster-management` of managed cluster by scaling deployment and setting replicas=0，even though all the pods are gone，the ManagedCluster CR status in hub doesn't change at all and `AVAILABLE` condition keeps `True` when running `kubectl get managedcluster -w`. Is this the correct way to unregister a managed cluster from hub on managed cluster?

My second try：
Scaling klusterlet pod in namespace `open-cluster-management` of managed cluster into 1 for recovery firstly.
I delete ManagedCluster CR in hub cluster by running `kubectl delete managedcluster <clusterName>`. It seems that the ManagedCluster CR itself and some role/rolebindings are removed，but takes no effect to other resources（cluster namespace、menifestwork in cluster namespace、resources in managed cluster etc.）. Is this the correct way to unregister a managed cluster from hub on hub cluster?

My third try：
Since ManagedCluster CR was deleted， I run `clusteradm unjoin --clustername <clusterName>` in managed cluster. It tells me to delete manifectworks in hub cluster firstly.
After running `kubectl delete manifectwork mw-01 -n <clusterNamespace>` on hub cluster，it stuck for a long time and not work. Both the manifectworks and the k8s resources in managed cluster are not deleted. It may be relevant to the lost of ManagedCluster and role/rolebindings, I guess. If so, when we run `kubectl delete managedcluster`, should OCM check the dependencies and tell us to delete the resource firstly？

In short, here are my confused questions :
What is the correct way(or official way) to unregister cleanly a managed cluster?
Is there any ways on hub cluster rather than on managed cluster to unregister cleanly and conveniently?

**Comment by ycyaoxdu** [comment]
@USER0308 hi, The suggested process to unregister a managed cluster from hub is:
1. Make sure no works running. You should not unjoin a managed cluster while it is running work which distributed by hub;
2. Unjoin the managedcluster from hub. Run `clusteradm unjoin` command on the managed cluster, this command will remove the OCM components on the managed cluster;
3. Delete the managedcluster CR on hub. The unjoin command just remove the resources on the managed cluster. so the resource on hub should be removed manually.
---

For question1: your third try is nearly correct, but you've deleted the managedcluster CR too early.
For question2: we haven't have a way to unregister a managed cluster on the hub cluster yet. you should log onto the managed cluster to unregister. And we're keep on making OCM better and easier.

---
Don't know whether the results answered your questions. By the way, any idea and contribution is welcomed! :)

**Comment by USER0308** [comment]
Thank you very much! @ycyaoxdu  I learn a lot from your answer. Thanks again  ^_^
